File: LC/python/Implement strStr().py
class Solution:
    # @param haystack, a string
    # @param needle, a string
    # @return an integer
    def strStr(self, haystack, needle):
        num = 0 
        n = 0
        i = 0 
        if len(haystack) < len(needle):
            return -1
        elif needle == '':
            return 0
            
        while (i < len(haystack) and num < len(needle)):
            if haystack[i] == needle[num]:
                n += 1
                if n == len(needle):
                    return i-len(needle)+1
                i +=1
                num +=1
            else:
                i = i -n+1
                num = 0
                n=0
        return -1
                
            
        

File: LC/java/Implement strStr().java
public class Solution {
    public int strStr(String haystack, String needle) {
        int m = haystack.length(), n = needle.length();
        String sub;
        
        if(m<n){
            return -1;
        }
        else if(haystack.equals(needle)){
            return 0;
        }
        else if(n == 0) {
            return 0;
        }
        
        for(int i = 0; i< m-n+1; i++){
            sub = haystack.substring(i,i+n);
            if (sub.equals(needle)){
                return i;
            }
        }
        return -1;
        
    }
}